File: app/src/main/java/workshop1024/com/xproject/controller/fragment/home/news/SubscribeNewsFragment.java
package workshop1024.com.xproject.controller.fragment.home.news;

import android.os.Bundle;

import workshop1024.com.xproject.R;
import workshop1024.com.xproject.controller.fragment.SubFragment;

public class SubscribeNewsFragment extends NewsListFragment implements SubFragment {
    private static final String SUBSCRIBE_NAME = "subscribe_Name";
    private String mSubscribeName;

    public static SubscribeNewsFragment newInstance(String subscribeName) {
        SubscribeNewsFragment subscribeNewsFragment = new SubscribeNewsFragment();
        subscribeNewsFragment.setNavigationItemId(R.id.leftnavigator_menu_home);
        Bundle args = new Bundle();
        args.putString(SUBSCRIBE_NAME, subscribeName);
        subscribeNewsFragment.setArguments(args);
        return subscribeNewsFragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mSubscribeName = getArguments().getString(SUBSCRIBE_NAME);
        }
    }

    @Override
    public void getNewsList() {
        mNewsRepository.getNewsListBySubscribe(mSubscribeName, this);
    }
}

File: app/src/main/java/workshop1024/com/xproject/controller/fragment/home/news/SubscribeNewsFragment.kt
package workshop1024.com.xproject.controller.fragment.home.news

import android.os.Bundle

import workshop1024.com.xproject.R
import workshop1024.com.xproject.controller.fragment.SubFragment

class SubscribeNewsFragment : NewsListFragment(), SubFragment {
    private var mSubscribeName: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
            mSubscribeName = arguments?.getString(SUBSCRIBE_NAME)
    }

    override fun getNewsList() {
        mSubscribeName?.let { mNewsRepository?.getNewsListBySubscribe(it, this) }
    }

    companion object {
        private const val SUBSCRIBE_NAME = "subscribe_Name"

        fun newInstance(subscribeName: String): SubscribeNewsFragment {
            val subscribeNewsFragment = SubscribeNewsFragment()
            subscribeNewsFragment.mNavigationItemId = R.id.leftnavigator_menu_home
            val args = Bundle()
            args.putString(SUBSCRIBE_NAME, subscribeName)
            subscribeNewsFragment.arguments = args
            return subscribeNewsFragment
        }
    }
}
